### DISCUSSION TITLE
Is it possible to receive dmx on one uart and copy/write it on another uart ?

### DISCUSSION THREAD

**Discussion by cgiles** [someweisguy/esp_dmx#159] [Q&A] [answered]
For my project, I would need to read on one uart and then copy on the other, is it possible, or would it fail ?

**Comment by bensuffolk** [comment, marked as answer]
That will work fine, I do that on one of my projects.

**Comment by TinyuZhao** [comment]
Have the same problem

**Comment by someweisguy** [comment]
As @bensuffolk says, it should work just fine.

I can't recall if this is mentioned in the documentation, but I will take a look and see if there is anything I can clarify! :)

**Comment by futesta** [comment]
Hey guys,
i would like to participate in the discussion.
@cgiles `do you find a solution and did it work for you?
I have the same problem. I'm trying to use UART1 and UART2 on an ESP32. Both work with the basic implementation Serial1.begin() or Serial2.begin(). Also at the same time. But when using the DMX lib the ESP32 crsahes at boot.
@bensuffolk can you possibly provide a working example?
Mike

**Reply by bensuffolk** [reply]
I’m on my phone right now, but that sounds like you are using the Arduino framework. I don’t have any experience using that with this library as I use ESP-IDF. In general it give you way more control over the ESP (for pretty much everything) to use ESP-IDF as it’s a professional dev framework as opposed to a hobby one, but I appreciate it’s a steeper learning curve.

**Reply by futesta** [reply]
Hey,

thanks for your fast reply, i appreciate that, take your call ;-)

Yes, I am currently "still" using the Arduino IDE for this project, but that is more out of convenience because I already had a working project and had actually only changed the ESP32 board...

In the meantime I was able to isolate the error and it works for now, the "bug" has already been described and discussed here: [#150](https://github.com/someweisguy/esp_dmx/issues/150)

I am currently using esp_dmx 4.1.0, Arduino 2.3.2 with the ESP 3 (3.0.4) Framework

Cheers
